src/main/java/bio/terra/service/dataset/StorageResourceDao.java

```diff
@@ -0,0 +1,97 @@
+package bio.terra.service.dataset;
+
+import bio.terra.app.model.GoogleRegion;
+import bio.terra.model.CloudPlatform;
+import bio.terra.app.configuration.DataRepoJdbcConfiguration;
+import bio.terra.common.DaoKeyHolder;
+import bio.terra.app.model.GoogleCloudResource;
+import bio.terra.service.dataset.exception.InvalidStorageException;
+import bio.terra.service.dataset.exception.StorageResourceNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Repository
+public class StorageResourceDao {
+
+    private static final String STORAGE_COLUMNS = "id, dataset_id, cloud_platform, " +
+        "cloud_resource, region ";
+    private static final String SQL_GET = "SELECT " + STORAGE_COLUMNS +
+        "FROM storage_resource WHERE dataset_id = :dataset_id";
+    private static final Logger logger = LoggerFactory.getLogger(StorageResourceDao.class);
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public StorageResourceDao(DataRepoJdbcConfiguration jdbcConfiguration) {
+        jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+    public List<StorageResource> getStorageResourcesByDatasetId(UUID datasetId) {
+        try {
+            MapSqlParameterSource params = new MapSqlParameterSource()
+                .addValue("dataset_id", datasetId);
+            return jdbcTemplate.query(SQL_GET, params, new StorageResourceMapper());
+        } catch (EmptyResultDataAccessException ex) {
+            throw new StorageResourceNotFoundException("Storage resource not found for dataset: "
+                + datasetId.toString());
```

Comment:
unnecessary `toString()`
```suggestion
                + datasetId);
```